README.md

```diff
@@ -207,18 +208,37 @@ Example of a partial CR:
 ```
 
 
-### Mounting binding information
+### Mounting and injecting binding information
+
+This specification allows for data to be mounted using volumes or injected using environment variables.  The best practice is to mount any sensitive information, such as passwords, since that will avoid accidently exposure via environment dumps and subprocesses. 
+
+The decision to mount vs inject is made in the following ascending order of precedence:
+* value of the `bindAs` attribute in the backing service as defined in its [annotations](annotations.md#data-model--building-blocks-for-expressing-binding-information), applying to the binding item referenced by the annotation.
+* value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
+* value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
+
+#### Injecting data
+
+The key `SERVICE_BINDINGS` acts as a global map of the service bindings and **MUST** always be injected into the environment.  It contains a JSON payload with a simple map, where the key points to an available service binding item and the value equals either `envVar` or `volume:<path>`.  Example:
+
+```
+{ "KAFKA_USERNAME": "envVar", "KAFKA_PASSWORD": "volume:/platform/bindings/secret/" }
```

Comment:
What are your thoughts about a JSON payload similar to this one:

```json
{
  "secretKeys": [
    {
      "name": "KAFKA_USERNAME",
      "type": "envVar"
    },
    {
      "name": "KAFKA_PASSWORD",
      "type": "volume",
      "mountPath": "/platform/bindings/secret/"
    }
  ]
}
```

The benefit is that we can extend it in the future if adding more info is needed.

Reply:
That's a good point @navidsh - we could even explore merging the ConfigMap metadata here in the future.

Perhaps using `bindAs` instead of `type`, to stay consistent with the annotations.

Reply:
added a new commit.